lib/cretonne/Cargo.toml

```diff
@@ -9,6 +9,13 @@ repository = "https://github.com/stoklund/cretonne"
 publish = false
 build = "build.rs"
 
+[features]
+# cretonne supports `no_std`
+# Using `std` is default, however
+default = ["std"]
+# workaround cargo not letting you conditionally *not* include dependencies
+std = ["hashmap_core/disable"]
```

Comment:
Would you mind expanding this comment a little to explain what hashmap_core/disable is?

Reply:
Sure. I'll explain it here as well.

Cargo doesn't have the ability to remove or disable a dependency when a feature is enabled. Since the convention is to have a default `std` feature, this was the only way that I could figure out how to get it to work.

I submitted a pull-request to `hashmap_core` that put the entire crate between a `disable` feature-gate. So, when `std` is enabled, it can turn on the disabled feature in hashmap_core so we don't pull in any code that we don't need. Ideally, I'd either like to integrate hashmap/hashset into liballoc, but atm that's not really an option.

If you have a better way of doing this that doesn't break convention, I'd be grateful to get rid of this hack.